### PR TITLE
*: add a ctx parameter to distsql.Select, kv.Client and tikv.Client

### PR DESCRIPTION
**Pull request by tiancaiamao**
With the support context, we can cancel the computation later.

### CODE REVIEW

File: executor/distsql.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"sync"
 	"time"
 
+	goctx "context"
```
Comment (hanfei1991) [inline executor/distsql.go]: move this to line 17?

**Comment by shenli** [comment]
LGTM

File: distsql/distsql.go [hunk 2]
```diff
@@ -14,6 +14,7 @@
 package distsql
 
 import (
+	"context"
```
Comment (coocood) [inline distsql/distsql.go]: goctx

File: kv/kv.go [hunk 3]
```diff
@@ -14,6 +14,7 @@
 package kv
 
 import (
+	"context"
```
Comment (coocood) [inline kv/kv.go]: goctx

File: store/localstore/local_client.go [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package localstore
 
 import (
+	"context"
```
Comment (coocood) [inline store/localstore/local_client.go]: ditto

File: store/localstore/xapi_test.go [hunk 5]
```diff
@@ -14,6 +14,7 @@
 package localstore
 
 import (
+	"context"
```
Comment (coocood) [inline store/localstore/xapi_test.go]: ditto

**Pull request merged**